chore(github): split up github actions (#3332)
ionic-team/stencil: .github/workflows/README.md

> # Stencil Continuous Integration (CI)

Author: This readme is intentionally high level. I didn't wanna regurgitate the details of every job/step here. If there's something y'all have questions about lmk. An inline comment somewhere may be necessary

Continuous integration (CI) is an important aspect of any project, and is used to verify and validate the changes to the

> codebase work as intended, do not introduce regressions (bugs), and adhere to coding standards (e.g. formatting rules).

Reviewer: ```suggestion
codebase work as intended, to avoid introducing regressions (bugs), and to adhere to coding standards (e.g. formatting rules).
```

Author: 👍 fixed in https://github.com/ionic-team/stencil/pull/3332/commits/8f85fe5d6465af68c7989ae9d4ade23d80beb2df

a set schedule.

The workflows that are run as a part of Stencil's CI process are declared as YAML files, and are stored in the same
directory as this file.
Each workflow file is explained in greater depth in the [workflows section](#workflows) of this document.

## Workflows

This section describes each of Stencil's GitHub Actions workflows.
Generally speaking, workflows are designed to be declarative in nature.
This section does not intend to duplicate the details of each workflow, but rather give a high level overview of each
one and mention nuances of each.

### Main (`main.yml`)

The main workflow for Stencil can be found in `main.yml` in this directory.
It is responsible for:
- Building Stencil (and Validating the Build)
- Running Unit, E2E, etc. Tests against the Build
- Verifying that the Codebase is Properly Formatted

Each of these tasks are codified as _jobs_, and are listed in `main.yml`.
Most of the jobs above are contingent on the 'build step' finishing (otherwise there would be nothing to test).
The diagram below displays the dependencies between each job.

> ```mermaid

Author: Use 'display rich diff' in GH to see the 🧜‍♀️ diagram
